scenarios/kubernetes_e2e.py

```diff
@@ -0,0 +1,304 @@
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Need to figure out why this only fails on travis
+# pylint: disable=bad-continuation
+
+"""Runs kubernetes e2e test with specified config"""
+
+import argparse
+import os
+import signal
+import subprocess
+import sys
+
+ORIG_CWD = os.getcwd()  # Checkout changes cwd
+
+def test_infra(*paths):
+    """Return path relative to root of test-infra repo."""
+    return os.path.join(ORIG_CWD, os.path.dirname(__file__), '..', *paths)
+
+
+def check_output(*cmd):
+    """Log and run the command, return output, raising on errors."""
+    print >>sys.stderr, 'Run:', cmd
+    return subprocess.check_output(cmd)
+
+
+def check(*cmd):
+    """Log and run the command, raising on errors."""
+    print >>sys.stderr, 'Run:', cmd
+    subprocess.check_call(cmd)
+
+
+def continuation_lines(fin):
+    """Reads in lines respect \\ in the end."""
+    for line in fin:
+        line = line.rstrip('\n')
+        while line.endswith('\\'):
+            line = line[:-1] + next(fin).rstrip('\n')
+        yield line
+
+
+def load_env(envpath):
+    """Load a env file into os.environ."""
+    if not os.path.isfile(envpath):
+        print >>sys.stderr, 'envfile %s does not exist' % envpath
+        return
+    with open(envpath, 'r') as envfile:
+        for line in continuation_lines(envfile):
+            tup = line.strip().split('=', 1)
+            if len(tup) == 2:
+                os.environ[tup[0].strip()] = tup[1].strip()
+
+
+def load_template(template_path, args):
+    """Load a templated env file into os.environ"""
+
+    env_map = {
+        '${version_client}' : args.version_client,
+        '${version_cluster}' : args.version_cluster,
+        '${version_old}' : args.version_old,
+        '${version_new}' : args.version_new,
+        '${image_old}' : args.image_old,
+        '${image_new}' : args.image_new
+    }
+
+    if not os.path.isfile(template_path):
+        print >>sys.stderr, 'envfile %s does not exist' % template_path
+        return
+    with open(template_path, 'r') as envfile:
+        for line in continuation_lines(envfile):
+            # Resolve vars
+            for key, val in env_map.items():
+                if key in line and val:
+                    line = line.replace(key, val)
+            tup = line.strip().split('=', 1)
+            if len(tup) == 2:
+                os.environ[tup[0].strip()] = tup[1].strip()
+
+
+def sigterm_handler(_signo, _frame):
+    """Handle signal.SIGTERM."""
+    print 'SIGTERM signo = %s, frame = %s' % (_signo, _frame)
+    check(['docker', 'stop', CONTAINER])
+
+
+def sigint_handler(_signo, _frame):
```

Comment:
Why do you have two functions that do exactly the same thing? Especially since whether it is handling a SIGTERM or SIGINT signal is captured in _signo.

Also prefixing method arguments with a underscore looks odd